dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/snowflake/SnowflakeTaskUtil.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.snowflake;
+
+import com.google.edwmigration.dumper.application.dumper.task.AbstractJdbcTask;
+import com.google.edwmigration.dumper.application.dumper.task.JdbcSelectTask;
+import com.google.edwmigration.dumper.application.dumper.task.Summary;
+
+final class SnowflakeTaskUtil {
+
+  static AbstractJdbcTask<Summary> fromVariant(
```

Comment:
Why do we create variants and convert them to tasks separately? It seems it's just extra complexity. Can't we have a TaskFactory that will create JdbcSelectTask with or without filter?

Reply:
Acknowledged, will fix in future PR

The original authors wanted to have a class that represents a task but without the main part of the query - this main part is provided by the `format` argument. For now I only moved this class from the connector to a separate file and made it AutoValue, but in the long term I also would like to remove TaskVariant.

Reply:
I removed `TaskVariant`